### PR TITLE
refactor: change private attr and step getter

### PR DESCRIPTION
**Pull request by mikita-sakalouski**
## Description
Adjust Output class and fix getting of attributes from Step classes.

## Related Issue
#81 

## Motivation and Context
Get correct behavior for class attributes and also raise Exception in proper manner

## How Has This Been Tested?
Existing tests

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Review by dannymeijer — APPROVED** [review]
Looks good

File: src/koheesio/models/reader.py [hunk 1]
```diff
@@ -2,13 +2,11 @@
...
 
+from typing import Optional
 from abc import ABC, abstractmethod
-from typing import Optional, TypeVar
 
 from koheesio import Step
-
-# Define a type variable that can be any type of DataFrame
-DataFrameType = TypeVar("DataFrameType")
+from koheesio.spark import DataFrame
```
Comment (dannymeijer) [inline src/koheesio/models/reader.py:9]: I'd like to not have Spark in the BaseReader - this is supposed to work without any trace of Spark

File: src/koheesio/spark/transformations/drop_column.py [hunk 2]
```diff
@@ -45,6 +45,6 @@ class DropColumn(ColumnsTransformation):
     In this example, the `product` column is dropped from the DataFrame `df`.
     """
 
-    def execute(self) -> ColumnsTransformation.Output:
-        self.log.info(f"{self.column=}")
+    def execute(self) -> None:
```
Comment (dannymeijer) [inline src/koheesio/spark/transformations/drop_column.py:48]: execute never returns None - this should be some form of Output

File: src/koheesio/spark/writers/delta/stream.py [hunk 3]
```diff
@@ -32,5 +32,7 @@ class Options(BaseModel):
     def execute(self) -> DeltaTableWriter.Output:
         if self.batch_function:
             self.streaming_query = self.writer.start()
+        # elif self.streaming and self.is_remote_spark_session:
+        #     self.streaming_query = self.writer.start()
```
Comment (dannymeijer) [inline src/koheesio/spark/writers/delta/stream.py:36]: Is this required?

**Review by dannymeijer — CHANGES_REQUESTED** [review]

File: src/koheesio/models/reader.py (hunk 1 above)
Comment (mikita-sakalouski) [reply src/koheesio/models/reader.py]: Fixed

File: src/koheesio/spark/transformations/drop_column.py (hunk 2 above)
Comment (mikita-sakalouski) [reply src/koheesio/spark/transformations/drop_column.py]: Fixed

File: src/koheesio/spark/writers/delta/stream.py (hunk 3 above)
Comment (mikita-sakalouski) [reply src/koheesio/spark/writers/delta/stream.py]: Fixed

**Review by dannymeijer — APPROVED** [review]

**Pull request merged**